conformance_tests/sysman/test_sysman_frequency/src/test_sysman_frequency.cpp

```diff
@@ -408,6 +412,42 @@ TEST_F(
     }
   }
 }
+
+TEST_F(
+    FREQUENCY_TEST,
+    GivenInvalidFrequencyRangeWhenRequestingSetFrequencyThenExpectMinAndMaxFrequencyInGetFrequencyCall) {
+  for (auto device : devices) {
+    uint32_t p_count = 0;
+    auto pfreq_handles = lzt::get_freq_handles(device, p_count);
+    if (p_count == 0) {
+      FAIL() << "No handles found: "
+             << _ze_result_t(ZE_RESULT_ERROR_UNSUPPORTED_FEATURE);
+    }
+
+    for (auto pfreq_handle : pfreq_handles) {
+      EXPECT_NE(nullptr, pfreq_handle);
+
+      zes_freq_range_t freq_range = {};
+      zes_freq_range_t modified_freq_range = {};
+
+      zes_freq_properties_t freq_property =
+          lzt::get_freq_properties(pfreq_handle);
+
+      if (freq_property.min - 100 >= 0) {
+        freq_range.min = freq_property.min - 100;
+      } else {
+        freq_range.min = 0;
+      }
+      freq_range.max = freq_property.max + 100;
+
+      lzt::set_freq_range(pfreq_handle, freq_range);
```

Comment:
we also have to check if get_range and set_range is supported or not for a particular domain. For example, for memory domain, freq range is not supported on windows. Also, if min and max returned from getRange are negative, they are not supported, so we should omit the checks/sets.

Reply:
please refer https://spec.oneapi.io/level-zero/latest/sysman/api.html#zes-freq-range-t[](url)

Reply:
I am checking if controlling freq range is supported using **canControl** field. Now for memory domain set range and get range won't be happening. I hope the canControl field will be false if set range is not supported.